innodb/src/main/java/org/apache/calcite/adapter/innodb/InnodbFilter.java

```diff
@@ -96,4 +98,9 @@ public static InnodbFilter create(RelOptCluster cluster, RelTraitSet traitSet,
   public RelCollation getImplicitCollation() {
     return indexCondition.getImplicitCollation();
   }
+
+  @Override public RexNode getCondition() {
```

Comment:
InnodbFIlter's condition contains both indexed and non-indexed conditions, but only the indexed filter will be executed. This will cause the FilterReduceExpression to produce incorrect results.